services/web/client/source/class/qxapp/desktop/PanelView.js

```diff
@@ -0,0 +1,161 @@
+/* ************************************************************************
+
+   qxapp - the simcore frontend
+
+   https://osparc.io
+
+   Copyright:
+     2018 IT'IS Foundation, https://itis.swiss
+
+   License:
+     MIT: https://opensource.org/licenses/MIT
+
+   Authors:
+     * Ignacio Pascual (ignapas)
+
+************************************************************************ */
+/* eslint-disable no-use-before-define */
+
+/**
+ * Display widget with a title bar and collapsible content.
+ */
+qx.Class.define("qxapp.desktop.PanelView", {
+  extend: qx.ui.core.Widget,
+
+  construct: function(title, content) {
+    this.base(arguments);
+
+    // Layout
+    const layout = new qx.ui.layout.VBox();
+    this._setLayout(layout);
+
+    // Title bar
+    this.__titleBar = new qx.ui.container.Composite(new qx.ui.layout.HBox(5))
+      .set({
+        appearance: "titlebar"
+      });
+    this._add(this.__titleBar);
+
+    if (title) {
+      this.setTitle(title);
+    }
+
+    // Content
+    if (content) {
+      content.setAppearance("panelview-content");
+      this.setContent(content);
+    }
+
+    // Attach handlers
+    this.__attachEventHandlers();
+  },
+
+  properties: {
+
+    title: {
+      check: "String",
+      nullable: true,
+      apply: "_applyTitle"
+    },
+
+    content: {
+      check: "qx.ui.core.Widget",
+      nullable: true,
+      apply: "_applyContent"
+    },
+
+    contentVisibility: {
+      init: true,
+      check: "Boolean",
+      apply: "_applyContentVisibility"
+    }
+  },
+
+  members: {
+
+    __titleBar: null,
+    __titleLabel: null,
+    __caret: null,
+
+    toggleContentVisibility: function() {
+      this.setContentVisibility(!this.getContentVisibility());
+    },
+
+    _applyContentVisibility: function(isVisible) {
+      if (this.getContent()) {
+        this.__caret.setSource(this.getContentVisibility() ? lessCaret : moreCaret);
+        if (isVisible) {
+          // BUGGY. Not using setTimeout causes weird rendering issue in QX. Using it causes a small flicker.
+          this.getContent().setVisibility("visible");
+          setTimeout(() => qx.bom.element.Animation.animateReverse(this.getContent().getContentElement()
+            .getDomElement(),
+          contentTransition(this.getContent().getContentElement()
+            .getDomElement().style.height)), 0);
+        } else {
+          qx.bom.element.Animation.animate(this.getContent().getContentElement()
+            .getDomElement(),
+          contentTransition(this.getContent().getContentElement()
+            .getDomElement().style.height));
+          setTimeout(() => this.getContent().setVisibility("excluded"), contentTransition().duration - 10);
+        }
```

Comment:
As discussed over lunch ... I would strongly suggest to drop the animation OR use the approach using css transitions applied by switching the decorator as shown in https://github.com/ITISFoundation/osparc-simcore/blob/9e61ac73c8e934cfa1a02b09fcde568521a8c798/services/web/client/source/class/qxapp/desktop/PrjEditor.js

Reply:
Transitions removed for the moment!